data_steward/vocabulary.py

```diff
@@ -1,11 +1,179 @@
-CONCEPT = 'concept'
-CONCEPT_ANCESTOR = 'concept_ancestor'
-CONCEPT_CLASS = 'concept_class'
-CONCEPT_RELATIONSHIP = 'concept_relationship'
-CONCEPT_SYNONYM = 'concept_synonym'
-DOMAIN = 'domain'
-DRUG_STRENGTH = 'drug_strength'
-RELATIONSHIP = 'relationship'
-VOCABULARY = 'vocabulary'
-VOCABULARY_TABLES = [CONCEPT, CONCEPT_ANCESTOR, CONCEPT_CLASS, CONCEPT_RELATIONSHIP, CONCEPT_SYNONYM, DOMAIN,
-                     DRUG_STRENGTH, RELATIONSHIP, VOCABULARY]
+import os
+import csv
+import re
+import sys
+import warnings
+
+from resources import CONCEPT, VOCABULARY, AOU_GENERAL_PATH, AOU_GENERAL_CONCEPT_CSV_PATH, hash_dir
+
+csv.field_size_limit(sys.maxsize)
+
+DELIMITER = '\t'
+LINE_TERMINATOR = '\n'
+RAW_DATE_REGEX = r'\d{8}$'  # yyyymmdd
+BQ_DATE_REGEX = r'\d{4}-\d{2}-\d{2}$'  # yyyy-mm-dd
+
+TRANSFORM_FILES = 'transform_files'
+APPEND_VOCABULARY = 'append_vocabulary'
+APPEND_CONCEPTS = 'append_concepts'
+ADD_AOU_GENERAL = 'add_aou_general'
+ERRORS = 'errors'
+
+AOU_GEN = 'AoU_General'
+AOU_GEN_VOCABULARY_CONCEPT_ID = '2000000000'
+AOU_GEN_VOCABULARY_REFERENCE = 'https://docs.google.com/document/d/10Gji9VW5-RTysM-yAbRa77rXqVfDfO2li2U4LxUQH9g'
+OMOP_VOCABULARY_CONCEPT_ID = '44819096'
+
+ERROR_APPENDING = 'Appending to {in_path} which already contains rows for ' + AOU_GEN
+
+
+def format_date_str(s):
+    """
+    Format a date string to yyyymmdd if it is not already
+    :param s: the date string
+    :return: the formatted date string
+    """
+    if re.match(BQ_DATE_REGEX, s):
+        return s
+    elif re.match(RAW_DATE_REGEX, s):
+        parts = s[0:4], s[4:6], s[6:8]
+        return '-'.join(parts)
+    else:
+        raise ValueError('Cannot parse value {v} as date'.format(v=s))
+
+
+def _transform_csv(in_fp, out_fp, err_fp=None):
+    if not err_fp:
+        err_fp = sys.stderr
+    csv_reader = csv.reader(in_fp, delimiter=DELIMITER)
+    header = next(csv_reader)
+    date_indexes = []
+    for i in range(0, len(header)):
```

Comment:
Codacy is complaining about this loop.  It could be changed to:

`for index, item in enumerate(header):`
`    if item.endswith('_date'):`
`        date_indexes.append(index)`